### PR TITLE
[FEATURE] Add pagination/search to CLI batch request listing

### PR DESCRIPTION
**Pull request by cdkini**
Please annotate your PR title to describe what the PR does, then give a brief bulleted description of your PR below. PR titles should begin with [BUGFIX], [FEATURE],  [DOCS], or [MAINTENANCE]. If a new feature introduces breaking changes for the Great Expectations API or configuration files, please also add [BREAKING]. You can read about the tags in our [contributor checklist](https://docs.greatexpectations.io/en/latest/contributing/contribution_checklist.html).

Changes proposed in this pull request:
- https://github.com/great-expectations/great_expectations/pull/3854

### Definition of Done
Please delete options that are not relevant.

- [x] My code follows the Great Expectations [style guide](https://docs.greatexpectations.io/en/latest/contributing/style_guide.html?highlight=style%20guide)
- [x] I have performed a [self-review](https://docs.greatexpectations.io/en/latest/contributing/contribution_checklist.html?highlight=checklist) of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added [unit tests](https://docs.greatexpectations.io/en/latest/contributing/testing.html#contributing-testing-writing-unit-tests) where applicable and made sure that new and existing tests are passing.
- [x] I have run any local integration tests and made sure that nothing is broken.

Thank you for submitting!

### CODE REVIEW

File: great_expectations/cli/batch_request.py
Comment (donaldheppner) [inline great_expectations/cli/batch_request.py]: why was this promoted to public? ... and where is the docstring? ;)

Comment (cdkini) [reply great_expectations/cli/batch_request.py]: It's responsibilities have grown considerably with the introduction of pagination and search so I thought it made sense to signify that through access modifiers.

It also makes testing that one component a lot easier (but I do realize that access modification to make testing is not the best practice).

I think the first point is still valid but let me know if you disagree and we can determine an alternative approach.

Comment (donaldheppner) [reply great_expectations/cli/batch_request.py]: When making something public, the question isn't about the amount of responsibility, it's about whether or not you expect other people to call it.

File: great_expectations/cli/batch_request.py
Comment (kenwade4) [inline great_expectations/cli/batch_request.py]: Maybe consider `user_selected_option.startswith('l')` for the check in case user puts `list` as the response

File: great_expectations/cli/batch_request.py
Comment (kenwade4) [inline great_expectations/cli/batch_request.py]: Ditto for `user_selected_option.startswith('s')` in case user puts `search` as the response

File: great_expectations/cli/batch_request.py
Comment (kenwade4) [inline great_expectations/cli/batch_request.py]: Consider the .startswith(..) here too

File: great_expectations/cli/batch_request.py
Comment (talagluck) [inline great_expectations/cli/batch_request.py:279]: ❤️

**Review by talagluck — COMMENTED** [review]
Looks great overall! I like Ken's suggestion for `startswith` and agree with Don about the public method, so I'll leave it to one of them to approve, but ping me if you would like me to approve as well.

File: great_expectations/cli/batch_request.py
Comment (cdkini) [reply great_expectations/cli/batch_request.py]: per discussion, keep this as a private function but still leaving docstring and function-specific tests

**Review by talagluck — APPROVED** [review]
LGTM - great work!

**Pull request merged**
